subsys/bluetooth/host/hci_core.c

```diff
@@ -849,6 +897,41 @@ static void le_remote_feat_complete(struct net_buf *buf)
 	bt_conn_unref(conn);
 }
 
+static void le_data_len_change(struct net_buf *buf)
+{
+	struct bt_hci_evt_le_data_len_change *evt = (void *)buf->data;
+	u16_t max_tx_octets = sys_le16_to_cpu(evt->max_tx_octets);
+	u16_t max_rx_octets = sys_le16_to_cpu(evt->max_rx_octets);
+	u16_t max_tx_time = sys_le16_to_cpu(evt->max_tx_time);
+	u16_t max_rx_time = sys_le16_to_cpu(evt->max_rx_time);
+	u16_t handle = sys_le16_to_cpu(evt->handle);
+	struct bt_conn *conn;
+
+	conn = bt_conn_lookup_handle(handle);
+	if (!conn) {
+		BT_ERR("Unable to lookup conn for handle %u", handle);
+		return;
+	}
+
+	BT_DBG("max. tx: %u (%uus), max. rx: %u (%uus)", max_tx_octets,
+	       max_tx_time, max_rx_octets, max_rx_time);
+
+	if (BT_FEAT_LE_PHY_2M(bt_dev.le.features) ||
+	    BT_FEAT_LE_PHY_CODED(bt_dev.le.features)) {
+		int err;
+
+		err = hci_le_set_phy(conn);
```

Comment:
No, legacy connection is established in 1M, then the switch has to happen.In LE Adv Ext, I think it can be established on the desired PHY on initiation.